lib/data/models/base/base_provider.dart

```diff
@@ -9,10 +9,15 @@ import 'package:retry/retry.dart';
 import '../../../widgets/atoms/server_error.dart';
 
 abstract class BaseProvider extends ChangeNotifier {
-  Widget runQuery<T extends BaseResult>({
-    required BaseOperation operation,
+  final GraphQLClient client;
+
+  BaseProvider({required this.client});
+
+  Widget runQuery({
```

Comment:
Really separate from the scope of this PR so including as a single comment: should we use [Flutter Hooks](https://medium.com/graphql-flutter/graphql-flutter-hooks-23ab5e77514e) rather than Widgets for API queries?

Reply:
Thanks for suggesting this, I need to look into this a bit more to be able to give an informed opinion.